### PR TITLE
Adding authorization checks for embeddable gadgets feature

### PR DESCRIPTION
**Pull request by Megala21**
Improvement for #203

### CODE REVIEW

File: apps/portal/controllers/gadgets.jag [hunk 1]
```diff
@@ -30,70 +29,116 @@ var pageId;
...
-        // Check whether the there is page in the dashboard with given page id
-        page = gadgets.findPage(dashboard, pageId);
-        // If the page is not available indicate error
-        if (!page) {
-            response.sendError(404, i18n.localize("page.not.found"));
-            return;
+        // If the user doesn't have any permission on the original dashboard, returns 401
+        var authorized = false;
+        // To view personalized dashboard, the user should have logged in, in the same domain and should have either editor or viewer permissions
+        if (user && user.domain == (urlDomain || superDomain) &&
```
Comment (sinthuja) [inline apps/portal/controllers/gadgets.jag]: Too big line, break into two lines.

File: apps/portal/js/gadget.js [hunk 2]
```diff
@@ -52,67 +24,19 @@ $(function () {
...
-        if (allPages.length > 0) {
-            page = (ues.global.page ? ues.global.page : allPages[0]);
-        }
-        for (var i = 0; i < allPages.length; i++) {
-            if (ues.global.page == allPages[i].id) {
-                page = allPages[i];
-            }
-        }
+        var page = ues.dashboards.findPage(ues.global.dashboard, ues.global.page);
+
```
Comment (sinthuja) [inline apps/portal/js/gadget.js]: Remove this blank line.

File: apps/portal/controllers/error-pages/error401.html [hunk 3]
```diff
@@ -17,7 +33,20 @@
 <i class="icon-unlink error-icon"></i>
 <h1>Error 401</h1>
 <h3>You do not have permission to access this page.Please contact your administrator and request permission.</h3>
-<h4><a href="/portal">Go to Portal</a></h4>
+<div id="portal" style="visibility: visible; display: block; ..">
+    <h4><a href="/portal">Go to Portal</a></h4>
+</div>
+<script type="text/javascript">
+    var link = document.getElementById('portal');
```
Comment (DilanUA) [inline apps/portal/controllers/error-pages/error401.html]: better be consistent with quotations for defining strings, use double quotes for outer strings and single quotes for inner strings.

File: apps/portal/controllers/error-pages/error401.html [hunk 4]
```diff
@@ -17,7 +33,20 @@
...
+    if (window.location.pathname.indexOf("/gadgets/") < 0) {
+        link.style.display = 'block';
+        link.style.visibility = 'visible';
+    }
```
Comment (DilanUA) [inline apps/portal/controllers/error-pages/error401.html]: unnecessary line break

File: apps/portal/controllers/error-pages/error404.html [hunk 5]
```diff
@@ -11,14 +27,26 @@
...
 <body>
     <i class="icon-unlink error-icon"></i>
     <h1>Error 404</h1>
     <h3>We can't find what you are looking for.</h3>
-    <h4><a href="/portal">Visit Portal</a></h4>
+    <div id="portal" style="visibility: visible; display: block; ..">
+        <h4><a href="/portal">Go to Portal</a></h4>
+    </div>
+    <script type="text/javascript">
+        var link = document.getElementById('portal');
```
Comment (DilanUA) [inline apps/portal/controllers/error-pages/error404.html]: better be consistent with quotations for defining strings, use double quotes for outer strings and single quotes for inner strings.

File: apps/portal/controllers/error-pages/error404.html [hunk 6]
```diff
@@ -11,14 +27,26 @@
...
+        if (window.location.pathname.indexOf("/gadgets/") < 0) {
+            link.style.display = 'block';
```
Comment (DilanUA) [inline apps/portal/controllers/error-pages/error404.html]: better be consistent with quotations for defining strings, use double quotes for outer strings and single quotes for inner strings.

File: apps/portal/controllers/error-pages/error404.html [hunk 7]
```diff
@@ -11,14 +27,26 @@
...
+            link.style.visibility = 'visible';
+        }
```
Comment (DilanUA) [inline apps/portal/controllers/error-pages/error404.html]: unnecessary line break

File: apps/portal/controllers/error-pages/error405.html [hunk 8]
```diff
@@ -17,7 +33,20 @@
 <i class="icon-unlink error-icon"></i>
 <h1>Error 405</h1>
 <h3>Method not allowed.</h3>
-<h4><a href="/portal">Go to Login Page</a></h4>
+<div id="portal" style="visibility: visible; display: block; ..">
+    <h4><a href="/portal">Go to Portal</a></h4>
+</div>
+<script type="text/javascript">
+    var link = document.getElementById('portal');
```
Comment (DilanUA) [inline apps/portal/controllers/error-pages/error405.html]: same applies here

File: apps/portal/controllers/error-pages/error500.html [hunk 9]
```diff
@@ -16,7 +32,20 @@
...
+    <script type="text/javascript">
+        var link = document.getElementById('portal');
+        if (window.location.pathname.indexOf("/gadgets/") < 0) {
+            link.style.display = 'block';
+            link.style.visibility = 'visible';
+        }
+        else {
+            link.style.visibility = 'hidden';
+            link.style.display = 'none';
+        }
```
Comment (DilanUA) [inline apps/portal/controllers/error-pages/error500.html]: same applies here

File: apps/portal/controllers/error-pages/error401.html (hunk 3 above)
Comment (Megala21) [inline apps/portal/controllers/error-pages/error401.html]: Thank you for pointing out. Error pages were refactored and these code segments were removed already.

File: apps/portal/controllers/error-pages/error401.html (hunk 4 above)
Comment (Megala21) [inline apps/portal/controllers/error-pages/error401.html]: Thank you for pointing out. Error pages were refactored and these code segments were removed already.

File: apps/portal/controllers/error-pages/error404.html (hunk 5 above)
Comment (Megala21) [inline apps/portal/controllers/error-pages/error404.html]: Thank you for pointing out. Error pages were refactored and these code segments were removed already.

File: apps/portal/controllers/error-pages/error404.html (hunk 7 above)
Comment (Megala21) [inline apps/portal/controllers/error-pages/error404.html]: Thank you for pointing out. Error pages were refactored and these code segments were removed already.

File: apps/portal/controllers/error-pages/error404.html (hunk 6 above)
Comment (Megala21) [inline apps/portal/controllers/error-pages/error404.html]: Thank you for pointing out. Error pages were refactored and these code segments were removed already.

File: apps/portal/controllers/error-pages/error500.html (hunk 9 above)
Comment (Megala21) [inline apps/portal/controllers/error-pages/error500.html]: Thank you for pointing out. Error pages were refactored and these code segments were removed already.

File: apps/portal/controllers/error-pages/error405.html (hunk 8 above)
Comment (Megala21) [inline apps/portal/controllers/error-pages/error405.html]: Thank you for pointing out. Error pages were refactored and these code segments were removed already.

**Pull request merged**